client/network/src/service.rs

```diff
@@ -201,12 +201,7 @@ impl<B: BlockT + 'static, H: ExHashT> NetworkWorker<B, H> {
 			params.chain.clone(),
 			params.protocol_id.clone(),
 			&params.network_config,
-			iter::once(Vec::new())
-				.chain(
-					(0..params.network_config.extra_sets.len() - 1)
-						.map(|_| default_notif_handshake_message.clone()),
-				)
-				.collect(),
+			vec![default_notif_handshake_message; params.network_config.extra_sets.len()],
```

Comment:
Well, no. elems[0] should be an empty Vec.

Reply:
Try this:

```
fn main() {
    core::iter::once(Vec::<()>::new()).inspect(|x| {dbg!(&x);}).collect::<Vec<_>>();
}
```

Reply:
Reverted. Interesting, didn't realize the code is not equivalent as before,  `[[]]` is not `[]`, https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=12e10b489de9b92a423279541eb5e317 . Do you mind sharing more about why elems[0] has to be an empty Vec?